ublox_neoM8/config_seabot_2024_10_08.txt
MON-VER - 0A 04 FA 00 45 58 54 20 43 4F 52 45 20 33 2E 30 31 20 28 31 31 31 31 34 31 29 00 00 00 00 00 00 00 00 30 30 30 38 30 30 30 30 00 00 52 4F 4D 20 42 41 53 45 20 32 2E 30 31 20 28 37 35 33 33 31 29 00 00 00 00 00 00 00 00 00 46 57 56 45 52 3D 54 49 4D 20 31 2E 31 30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 50 52 4F 54 56 45 52 3D 32 32 2E 30 30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 4D 4F 44 3D 4E 45 4F 2D 4D 38 54 2D 30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 46 49 53 3D 30 78 45 46 34 30 31 35 20 28 31 30 30 31 31 31 29 00 00 00 00 00 00 00 00 00 47 50 53 3B 47 4C 4F 3B 47 41 4C 3B 42 44 53 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 53 42 41 53 3B 49 4D 45 53 3B 51 5A 53 53 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
CFG-ANT - 06 13 04 00 00 00 F0 39
CFG-DAT - 06 06 02 00 00 00
CFG-GNSS - 06 3E 3C 00 00 20 20 07 00 08 10 00 01 00 01 01 01 01 03 00 01 00 01 01 02 04 08 00 01 00 01 01 03 08 10 00 00 00 01 01 04 00 08 00 00 00 01 03 05 00 03 00 00 00 01 05 06 08 0E 00 01 00 01 01
CFG-INF - 06 02 0A 00 00 00 00 00 00 00 00 00 00 00
CFG-INF - 06 02 0A 00 01 00 00 00 07 07 00 07 07 00
CFG-ITFM - 06 39 08 00 F3 AC 62 2D 1E 23 00 00
CFG-LOGFILTER - 06 47 0C 00 01 00 00 00 00 00 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0B 30 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0B 33 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0B 31 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0B 01 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 21 08 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 0B 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 09 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 02 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 06 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 07 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 21 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0A 08 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 60 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 22 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 36 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 31 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 04 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 61 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 39 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 09 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 34 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 01 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 02 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 07 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 35 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 32 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 06 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 03 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 30 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 24 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 25 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 23 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 20 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 26 00 FF 00 00 00 00
CFG-MSG - 06 01 08 00 01 21 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 01 11 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 01 12 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 02 61 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 02 14 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 02 15 00 01 00 00 00 00
CFG-MSG - 06 01 08 00 02 59 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 02 13 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 02 20 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0D 04 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0D 03 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0D 01 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 0D 06 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F0 00 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 01 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 02 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 03 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 04 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 05 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 06 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F0 07 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F0 08 01 00 00 01 01 00
CFG-MSG - 06 01 08 00 F0 09 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F0 0A 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F0 0D 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F0 0F 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F1 00 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F1 03 00 00 00 00 00 00
CFG-MSG - 06 01 08 00 F1 04 00 00 00 00 00 00
CFG-NAV5 - 06 24 24 00 FF FF 00 03 00 00 00 00 10 27 00 00 05 00 FA 00 FA 00 64 00 5E 01 00 3C 00 00 00 00 00 00 00 00 00 00 00 00
CFG-NAVX5 - 06 23 28 00 02 00 FF FF 7F 02 00 00 03 02 01 20 09 00 00 01 00 00 4B 07 00 01 00 00 01 01 00 00 00 64 64 00 00 01 11 00 00 00 00 00
CFG-NMEA - 06 17 14 00 00 40 00 02 00 00 00 00 00 00 00 01 00 00 00 00 00 00 00 00
CFG-ODO - 06 1E 14 00 00 00 00 00 00 00 00 00 19 46 19 66 0A 32 00 00 99 4C 00 00
CFG-PM2 - 06 3B 30 00 02 06 00 00 00 14 42 01 E8 03 00 00 10 27 00 00 00 00 00 00 00 00 00 00 2C 01 00 00 4F C1 03 00 86 02 00 00 FE 00 00 00 64 40 01 00 00 00 00 00
CFG-PRT - 06 00 14 00 00 00 00 00 84 00 00 00 00 00 00 00 07 00 03 00 00 00 00 00
CFG-PRT - 06 00 14 00 01 00 00 00 C0 08 00 00 00 C2 01 00 07 00 03 00 00 00 00 00
CFG-PRT - 06 00 14 00 03 00 00 00 00 00 00 00 00 00 00 00 07 00 03 00 00 00 00 00
CFG-PRT - 06 00 14 00 04 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
CFG-RATE - 06 08 06 00 E8 03 01 00 01 00
CFG-RINV - 06 34 18 00 00 4E 6F 74 69 63 65 3A 20 6E 6F 20 64 61 74 61 20 73 61 76 65 64 21 00
CFG-RXM - 06 11 02 00 48 00
CFG-SBAS - 06 16 08 00 01 03 03 00 00 00 00 00
CFG-TMODE2 - 06 3D 1C 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
CFG-TP5 - 06 31 20 00 00 01 00 00 32 00 00 00 40 42 0F 00 40 42 0F 00 A0 86 01 00 A0 86 01 00 00 00 00 00 77 00 00 00
CFG-TP5 - 06 31 20 00 01 01 00 00 32 00 00 00 04 00 00 00 01 00 00 00 48 E8 01 00 A0 86 01 00 00 00 00 00 FE 00 00 00
CFG-USB - 06 1B 6C 00 46 15 A8 01 00 00 00 00 64 00 02 00 75 2D 62 6C 6F 78 20 41 47 20 2D 20 77 77 77 2E 75 2D 62 6C 6F 78 2E 63 6F 6D 00 00 00 00 00 00 75 2D 62 6C 6F 78 20 47 4E 53 53 20 72 65 63 65 69 76 65 72 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00


pic_alim/config.asm

_init_io:

;config.c,56 :: 		void init_io(){
;config.c,59 :: 		ANSEL = 0b00001111;  // Set RC0,RC1,RC2,RC3 to analog (AN4,AN5,AN6,AN7)
	MOVLW       15
	MOVWF       ANSEL+0 
;config.c,60 :: 		ANSELH = 0b11110100; // Set RC6, RC7, RC5 to analog (AN8, AN9, AN11)
	MOVLW       244
	MOVWF       ANSELH+0 
;config.c,62 :: 		CM1CON0 = 0x00; // Not using the comparators
	CLRF        CM1CON0+0 
;config.c,63 :: 		CM2CON0 = 0x00;
	CLRF        CM2CON0+0 
;config.c,67 :: 		TRISA = 0xFF;
	MOVLW       255
	MOVWF       TRISA+0 
;config.c,68 :: 		TRISA2_bit = 1; // INPUT (ILS)
	BSF         TRISA2_bit+0, BitPos(TRISA2_bit+0) 
;config.c,69 :: 		TRISA5_bit = 0; // OUTPUT (POWER)
	BCF         TRISA5_bit+0, BitPos(TRISA5_bit+0) 
;config.c,71 :: 		INTCON2.RABPU = 0; // PORTA and PORTB Pull-up Enable bit
	BCF         INTCON2+0, 7 
;config.c,72 :: 		WPUA.WPUA2 = 1; // Pull-up enabled sur RA2
	BSF         WPUA+0, 2 
;config.c,74 :: 		TRISB = 0xFF;
	MOVLW       255
	MOVWF       TRISB+0 
;config.c,75 :: 		TRISB5_bit = 1; // RC5 input for AN11
	BSF         TRISB5_bit+0, BitPos(TRISB5_bit+0) 
;config.c,76 :: 		TRISB7_bit = 0; // OUTPUT (LED)
	BCF         TRISB7_bit+0, BitPos(TRISB7_bit+0) 
;config.c,78 :: 		TRISC = 0xFF;
	MOVLW       255
	MOVWF       TRISC+0 
;config.c,79 :: 		TRISC0_bit = 1; // RC0 input for AN4
	BSF         TRISC0_bit+0, BitPos(TRISC0_bit+0) 
;config.c,80 :: 		TRISC1_bit = 1; // RC1 input for AN5
	BSF         TRISC1_bit+0, BitPos(TRISC1_bit+0) 
;config.c,81 :: 		TRISC2_bit = 1; // RC2 input for AN6
	BSF         TRISC2_bit+0, BitPos(TRISC2_bit+0) 
;config.c,82 :: 		TRISC3_bit = 1; // RC3 input for AN7
	BSF         TRISC3_bit+0, BitPos(TRISC3_bit+0) 
;config.c,84 :: 		TRISC6_bit = 1; // RC6 input for AN8
	BSF         TRISC6_bit+0, BitPos(TRISC6_bit+0) 
;config.c,85 :: 		TRISC7_bit = 1; // RC7 input for AN9
	BSF         TRISC7_bit+0, BitPos(TRISC7_bit+0) 
;config.c,86 :: 		}
L_end_init_io:
	RETURN      0
; end of _init_io

_measure_power:

;config.c,88 :: 		void measure_power(){
;config.c,89 :: 		battery_voltage[0] = ADC_Get_Sample(4);   // Get 10-bit results of AD conversion AN4 batterie 1
	MOVLW       4
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _battery_voltage+0 
	MOVF        R1, 0 
	MOVWF       _battery_voltage+1 
;config.c,90 :: 		battery_voltage[1] = ADC_Get_Sample(5);   // Get 10-bit results of AD conversion AN5 batterie 2
	MOVLW       5
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _battery_voltage+2 
	MOVF        R1, 0 
	MOVWF       _battery_voltage+3 
;config.c,91 :: 		battery_voltage[2] = ADC_Get_Sample(6);   // Get 10-bit results of AD conversion AN6 batterie 3
	MOVLW       6
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _battery_voltage+4 
	MOVF        R1, 0 
	MOVWF       _battery_voltage+5 
;config.c,92 :: 		battery_voltage[3] = ADC_Get_Sample(7);   // Get 10-bit results of AD conversion AN7 batterie 4
	MOVLW       7
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _battery_voltage+6 
	MOVF        R1, 0 
	MOVWF       _battery_voltage+7 
;config.c,95 :: 		power_current[0] = ADC_Get_Sample(11);
	MOVLW       11
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _power_current+0 
	MOVF        R1, 0 
	MOVWF       _power_current+1 
;config.c,96 :: 		power_current[1] = ADC_Get_Sample(8);
	MOVLW       8
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _power_current+2 
	MOVF        R1, 0 
	MOVWF       _power_current+3 
;config.c,97 :: 		power_current[2] = ADC_Get_Sample(9);
	MOVLW       9
	MOVWF       FARG_ADC_Get_Sample_channel+0 
	CALL        _ADC_Get_Sample+0, 0
	MOVF        R0, 0 
	MOVWF       _power_current+4 
	MOVF        R1, 0 
	MOVWF       _power_current+5 
;config.c,98 :: 		}
L_end_measure_power:
	RETURN      0
; end of _measure_power

_init_timer0:

;config.c,100 :: 		void init_timer0(){
;config.c,101 :: 		T0CON = 0x85; // TIMER0 ON (1 s)
	MOVLW       133
	MOVWF       T0CON+0 
;config.c,102 :: 		TMR0H = TIMER0_CPT_H;
	MOVLW       11
	MOVWF       TMR0H+0 
;config.c,103 :: 		TMR0L = TIMER0_CPT_L;
	MOVLW       220
	MOVWF       TMR0L+0 
;config.c,104 :: 		TMR0IE_bit = 0;
	BCF         TMR0IE_bit+0, BitPos(TMR0IE_bit+0) 
;config.c,105 :: 		}
L_end_init_timer0:
	RETURN      0
; end of _init_timer0

_init_timer3:

;config.c,107 :: 		void init_timer3(){
;config.c,108 :: 		T3CON = 0x30;
	MOVLW       48
	MOVWF       T3CON+0 
;config.c,109 :: 		TMR3IF_bit = 0;
	BCF         TMR3IF_bit+0, BitPos(TMR3IF_bit+0) 
;config.c,110 :: 		TMR3H = TIMER3_CPT_H;
	MOVLW       60
	MOVWF       TMR3H+0 
;config.c,111 :: 		TMR3L = TIMER3_CPT_L;
	MOVLW       176
	MOVWF       TMR3L+0 
;config.c,112 :: 		TMR3IE_bit = 0;
	BCF         TMR3IE_bit+0, BitPos(TMR3IE_bit+0) 
;config.c,113 :: 		}
L_end_init_timer3:
	RETURN      0
; end of _init_timer3

_ils_analysis:

;config.c,115 :: 		void ils_analysis(unsigned short new_state){
;config.c,116 :: 		if(ILS==0){ // Magnet detected
	BTFSC       PORTA+0, 2 
	GOTO        L_ils_analysis0
;config.c,117 :: 		ils_cpt--;
	DECF        _ils_cpt+0, 1 
;config.c,118 :: 		set_led_on = 1;
	MOVLW       1
	MOVWF       _set_led_on+0 
;config.c,119 :: 		}
	GOTO        L_ils_analysis1
L_ils_analysis0:
;config.c,121 :: 		ils_cpt = ILS_CPT_TIME;
	MOVLW       4
	MOVWF       _ils_cpt+0 
;config.c,122 :: 		set_led_on = 0;
	CLRF        _set_led_on+0 
;config.c,123 :: 		ils_removed = 1;
	MOVLW       1
	MOVWF       _ils_removed+0 
;config.c,124 :: 		}
L_ils_analysis1:
;config.c,126 :: 		if(ils_removed == 1 && ils_cpt == 0){
	MOVF        _ils_removed+0, 0 
	XORLW       1
	BTFSS       STATUS+0, 2 
	GOTO        L_ils_analysis4
	MOVF        _ils_cpt+0, 0 
	XORLW       0
	BTFSS       STATUS+0, 2 
	GOTO        L_ils_analysis4
L__ils_analysis5:
;config.c,127 :: 		ils_cpt = ILS_CPT_TIME;
	MOVLW       4
	MOVWF       _ils_cpt+0 
;config.c,128 :: 		state = new_state;
	MOVF        FARG_ils_analysis_new_state+0, 0 
	MOVWF       _state+0 
;config.c,129 :: 		ils_removed = 0;
	CLRF        _ils_removed+0 
;config.c,130 :: 		set_led_on = 0;
	CLRF        _set_led_on+0 
;config.c,131 :: 		}
L_ils_analysis4:
;config.c,132 :: 		}
L_end_ils_analysis:
	RETURN      0
; end of _ils_analysis
